src/restapi.rs

```diff
@@ -1,12 +1,89 @@
 use crate::utils::DisplayVec;
-use goose::goose::{GooseUser, TransactionResult};
+use goose::goose::{GooseMethod, GooseRequest, GooseUser, TransactionError, TransactionResult};
+use reqwest::{Client, RequestBuilder};
+
+use crate::utils::GooseUserData;
+use rand::Rng;
 use serde_json::json;
 use std::sync::{
     Arc,
     atomic::{AtomicUsize, Ordering},
 };
 use urlencoding::encode;
 
+/// Get advisory total count
+pub async fn get_advisory_total(user: &mut GooseUser) -> TransactionResult {
+    let response = user.get("/api/v2/advisory").await?;
+    let json_data = response.response?.json::<serde_json::Value>().await?;
+
+    // Extract total from the response
+    if let Some(total) = json_data.get("total").and_then(|t| t.as_u64()) {
+        log::info!("Advisory total count: {}", total);
+        user.set_session_data(GooseUserData {
+            total_advisories: Some(total),
+            advisory_id: None,
+        });
+        return Ok(());
+    }
+    Err(Box::new(TransactionError::Custom(
+        "Failed to get advisory total count".to_string(),
+    )))
+}
+
+/// List advisory with random offset and limit=1
+pub async fn list_advisory_random_single(user: &mut GooseUser) -> TransactionResult {
+    // Get the total count first
+    let total = {
+        let goose_user_data = user
```

Comment:
The code for getting the user data is used quite often in the same way. Please extract this into an own function. As part of the `GooseUserData` struct.